gufe/tests/test_protocol.py

```diff
@@ -112,7 +112,7 @@ def _create(
         self,
         stateA: ChemicalSystem,
         stateB: ChemicalSystem,
-        mapping: Optional[dict[str, ComponentMapping]] = None,
+        mapping: Optional[list[ComponentMapping]] = None,
```

Comment:
Looking at the diff on this: https://github.com/OpenFreeEnergy/openfe/pull/666

Maybe we could go further and allow `Union[ComponentMapping, list[ComponentMapping]]`, i.e. one or more mappings.  It might be more intuitive to allow a single mapping to be provided, rather than more often than not a list with a single entry.